tests/runtime/ray/test_dag.py

```diff
@@ -36,81 +36,180 @@ def client():
-        (_client.WorkflowStatus.CANCELED, None, None, State.TERMINATED),
-        (_client.WorkflowStatus.CANCELED, TEST_TIME, None, State.TERMINATED),
-        (_client.WorkflowStatus.CANCELED, TEST_TIME, TEST_TIME, State.TERMINATED),
+        (_client.WorkflowStatus.SUCCESSFUL, start, end, task_states, State.SUCCEEDED)
+        for start in [None, TEST_TIME]
+        for end in [None, TEST_TIME]
+        for task_states in task_state_cases
+    ]
+    # Resumable: report as FAILED
+    + [
+        (_client.WorkflowStatus.RESUMABLE, start, end, task_states, State.FAILED)
+        for start in [None, TEST_TIME]
+        for end in [None, TEST_TIME]
+        for task_states in task_state_cases
+    ]
+    # Canceled: report as TERMINATED
+    + [
+        (_client.WorkflowStatus.CANCELED, start, end, task_states, State.TERMINATED)
+        for start in [None, TEST_TIME]
+        for end in [None, TEST_TIME]
+        for task_states in task_state_cases
+    ]
+    # Running:
+    # (defensive case for race conditions in Ray) status is running but no start time
+    # reported. Report as WAITING
+    + [
+        (_client.WorkflowStatus.RUNNING, None, None, task_states, State.WAITING)
+        for task_states in task_state_cases
+    ]
+    # the wf has a start time and no end time - it's definitely running.
+    + [
+        (_client.WorkflowStatus.RUNNING, TEST_TIME, None, task_states, State.RUNNING)
+        for task_states in task_state_cases
+    ]
+    # the wf has a start and end time, but the status is still being reported as
+    # 'running' - Ray hasn't updated the status yet. Report as SUCCEEDED.
+    + [
+        (
+            _client.WorkflowStatus.RUNNING,
+            TEST_TIME,
+            TEST_TIME,
+            task_states,
+            State.SUCCEEDED,
+        )
+        for task_states in task_state_cases
+    ]
+    # Failed:
+    # Tasks are still running, report as RUNNING
+    + [
+        (_client.WorkflowStatus.FAILED, start, end, task_states, State.RUNNING)
+        for start in [None, TEST_TIME]
+        for end in [None, TEST_TIME]
+        for task_states in [
+            [State.SUCCEEDED, State.RUNNING],
+            [State.TERMINATED, State.RUNNING],
+        ]
+    ]
+    # Tasks have all completed, report as FAILED
+    + [
+        (
+            _client.WorkflowStatus.FAILED,
+            start,
+            end,
+            [State.SUCCEEDED, State.FAILED],
+            State.FAILED,
+        )
+        for start in [None, TEST_TIME]
+        for end in [None, TEST_TIME]
     ],
 )
 def test_workflow_state_from_ray_meta(
     ray_wf_status,
     start_time,
     end_time,
+    task_states,
     expected_orq_status,
+    monkeypatch: pytest.MonkeyPatch,
 ):
+    mock_task_state: Mock = create_autospec(_dag._task_state_from_ray_meta)
+    mock_task_state.side_effect = task_states
+    mock_ray_meta = create_autospec(dict)
+    monkeypatch.setattr(_dag.__name__ + "._task_state_from_ray_meta", mock_task_state)
```

Comment:
It's identical to `monkeypatch.setattr("orquestra.sdk._ray._dag._task_state_from_ray_meta", mock_task_state)` , and is left over when when I _thought_ this test was failing because the patch wasn't working. Changed back to a more normal look [8c09a1a](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/317/commits/8c09a1ab304fc2e65bcf4bca90e22aad326bcbc7)